rasa_nlu/evaluate.py

```diff
@@ -502,60 +520,91 @@ def run_cv_evaluation(td, n_folds, nlu_config):
 
         utils.remove_model(model_dir)
 
     os.rmdir(os.path.join(tmp_dir, "default"))
     os.rmdir(tmp_dir)
 
-    return CVEvaluationResult(dict(train_results), dict(test_results))
+    return (CVEvaluationResult(dict(train_results), dict(test_results)),
+            CVEvaluationResult(dict(entity_train_results),
+                               dict(entity_test_results)))
 
 
-def compute_metrics(interpreter, corpus, results):
+def compute_metrics(interpreter, corpus, intent_results, entity_results):
     """Computes evaluation metrics for a given corpus and
 
     appends them to results.
     """
-    y = [e.get("intent") for e in corpus]
-
-    preds = []
-    for e in corpus:
-        res = interpreter.parse(e.text)
-        if res.get('intent'):
-            preds.append(res['intent'].get('name'))
-        else:
-            preds.append(None)
 
-    y = clean_intent_labels(y)
-    preds = clean_intent_labels(preds)
+    # y = [e.get("intent") for e in corpus]
+    intent_targets, entity_targets = get_targets(corpus)
+    intent_predictions, entity_predictions, tokens = get_predictions(interpreter,
+                                                                     corpus)
+    intent_targets, intent_predictions = remove_empty_intent_examples(
+                                            intent_targets, intent_predictions)
 
     # compute fold metrics
-    _, precision, f1, accuracy = get_evaluation_metrics(y, preds)
+    _, precision, f1, accuracy = get_evaluation_metrics(intent_targets,
+                                                        intent_predictions)
+
+    intent_results["Accuracy"].append(accuracy)
+    intent_results["F1-score"].append(f1)
+    intent_results["Precision"].append(precision)
+
+    extractors = get_entity_extractors(interpreter)
+    aligned_predictions = align_all_entity_predictions(entity_targets,
+                                                       entity_predictions,
+                                                       tokens, extractors)
+
+    merged_targets = merge_labels(aligned_predictions)
+    merged_targets = substitute_labels(merged_targets, "O", "no_entity")
+    for extractor in extractors:
+        merged_predictions = merge_labels(aligned_predictions, extractor)
+        merged_predictions = substitute_labels(merged_predictions, "O",
+                                               "no_entity")
+        _, precision, f1, accuracy = get_evaluation_metrics(merged_targets,
+                                                            merged_predictions)
+        entity_results[extractor]["Accuracy"].append(accuracy)
+        entity_results[extractor]["F1-score"].append(f1)
+        entity_results[extractor]["Precision"].append(precision)
+
+
+def return_results(results, dataset):
+    """Returns results of crossvalidation
+    :param results: dictionary of results returned from cv
+    :param dataset: string of which dataset the results are from, e.g.
```

Comment:
let's call this `dataset_name` or something - this looks a bit to much like it is a dataset